server/website/website/views.py

```diff
@@ -525,10 +525,16 @@ def handle_result_files(session, files):
         return HttpResponse("Result stored successfully!")
 
     result_id = result.pk
-    response = chain(aggregate_target_results.s(result.pk),
-                     map_workload.s(),
-                     configuration_recommendation.s()).apply_async()
-    taskmeta_ids = [response.parent.parent.id, response.parent.id, response.id]
+    if session.name == 'GPR':
```

Comment:
You may want to choose the algorithm based on the session.algorithm? 

https://github.com/cmu-db/ottertune/blob/b9dc726b9c498d200640e5e4921ded0fd8de2b5d/server/website/website/views.py#L535